### DISCUSSION TITLE
Upload file to Supabase storage using the REST api

### DISCUSSION THREAD

**Discussion by heloWelo** [supabase/supabase#22245] [Questions]
I'm having difficulty doing a simple file upload to a Storage Bucket.

https://myProjectID.supabase.co/storage/v1/object/stage1route/

I've added an API key and a Bearer token.

I'm using Postman that I thought might help but every permutation I try gives me a '400 Bad Request'

It's a .csv file and and I've tried different 'content-types'.

Not sure where I'm going wrong. Any ideas?

**Comment by GaryAustin1** [comment]
Any detail with the 400 error?  Maybe a message in the response?
Your path is OK except not clear what your /stageroute/ is.  It should be something like `/bucketname/pathtofile.csv`

**Comment by heloWelo** [comment]
Thanks Gary

Just managed to get it working. I've no idea what the error was but I went back to my original CURL, switched to binary within Postman and changed the Bearer key from the JWT key to the Secret Key. (I think AI lead me astray here...)

'stage1route' was my bucket name and I had a .csv file tagged on the end.

Thanks for your reply!
